### PR TITLE
Mobile ucr bug

### PR DESCRIPTION
**Pull request by proteusvacuum**
http://manage.dimagi.com/default.asp?218506#1098056

Looks like there is almost zero error handling in this code, so wasn't sure the best way of going about this. Decided to go the dragnet route. 

@czue

### CODE REVIEW

**Comment by czue** [comment]
lol what's the dragnet route?

File: corehq/apps/app_manager/views/modules.py [hunk 1]
```diff
@@ -631,13 +631,22 @@ def edit_report_module(request, domain, app_id, module_id):
     module = app.get_module(module_id)
     assert isinstance(module, ReportModule)
     module.name = params['name']
-    module.report_configs = [ReportAppConfig.wrap(spec) for spec in params['reports']]
+    try:
+        module.report_configs = [ReportAppConfig.wrap(spec) for spec in params['reports']]
+    except Exception:
+        return HttpResponseBadRequest(_("There was a problem processing your request."))
```
Comment (czue) [inline corehq/apps/app_manager/views/modules.py]: do we want to consider logging these at all? do you know what category of error you expect to find here?

File: corehq/apps/app_manager/views/modules.py [hunk 2]
```diff
@@ -631,13 +631,22 @@ def edit_report_module(request, domain, app_id, module_id):
...
     if (feature_previews.MODULE_FILTER.enabled(domain) and
             app.enable_module_filtering):
         module['module_filter'] = request.POST.get('module_filter')
     module.media_image.update(params['multimedia']['mediaImage'])
     module.media_audio.update(params['multimedia']['mediaAudio'])
-    app.save()
+
+    try:
+        app.save()
+    except Exception:
```
Comment (czue) [inline corehq/apps/app_manager/views/modules.py]: same question here

**Comment by czue** [comment]
does this actually solve the error or just show a more reasonable message?

**Comment by proteusvacuum** [comment]
this: https://github.com/dimagi/commcare-hq/commit/1d8dfb466c6135b2606b792e7323a47033557691 actually solves the error.

It seems like there isn't much error handling that happens in this area of the code, and I was wondering whether `except Exception` was a little to broad, but couldn't find any named exceptions up the chain to catch. (dragnet route: just catch everything).

One option is to go through and name a bunch of exceptions that might reasonably get thrown and show some nice error messages (which I thought might be out of scope, is preferable from a UX standpoint, but might require more context).

The other option could be to handle 500's better on the javascript side (i.e. not just dump responseText into the `alert_user` dialog).

**Comment by czue** [comment]
ah sneaky i missed that little change. your reasoning seems fine for now - we can revisit better error reporting if and when it comes up through interrupt/support/product

**Comment by proteusvacuum** [comment]
I do think it's a good idea to log these errors though as they'll just get swallowed right now. I'll add a follow up.

**Pull request merged**